File: viewpager/src/main/java/com/donfyy/viewpager/lazyloading/lazy2/MyFragmentPagerAdapter.java
package com.donfyy.viewpager.lazyloading.lazy2;

import java.util.List;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

/**
 * @author 享学课堂 Alvin
 * @package com.xiangxue.alvin.mylazyloadingfragment
 * @fileName MyFragmentPagerAdapter
 * @date on 2019/6/10
 * @qq 2464061231
 **/
public class MyFragmentPagerAdapter extends FragmentPagerAdapter {
    private static final String TAG = "MyFragmentPagerAdapter";

    private List<Fragment> fragmentList;
    public MyFragmentPagerAdapter(FragmentManager fm, List<Fragment> flist) {
        super(fm);
        fragmentList = flist;
    }
    @Override
    public Fragment getItem(int i) {
        return fragmentList.get(i);
    }

    @Override
    public int getCount() {
        return fragmentList.size();
    }
}

File: viewpager/src/main/java/com/donfyy/viewpager/lazyloading/lazy2/MyFragmentPagerAdapter.kt
package com.donfyy.viewpager.lazyloading.lazy2

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class MyFragmentPagerAdapter(fm: FragmentManager, private val fragmentList: List<Fragment>) : FragmentPagerAdapter(fm) {
    override fun getItem(i: Int): Fragment {
        return fragmentList[i]
    }

    override fun getCount(): Int {
        return fragmentList.size
    }

    companion object {
        private const val TAG = "MyFragmentPagerAdapter"
    }

}